src/auth/certificatesubjecttokensupplier.ts

```diff
@@ -0,0 +1,299 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GaxiosOptions} from 'gaxios';
+import {SubjectTokenSupplier} from './identitypoolclient';
+import {AuthClient} from './authclient';
+import {getWellKnownCertificateConfigFileLocation, isValidFile} from '../util';
+import * as fs from 'fs';
+import {X509Certificate} from 'crypto';
+import * as https from 'https';
+
+// --- Constants ---
```

Comment:
I noticed the use of sectioning comments like `// --- Main Class ---`. While these can be useful for organization, it seems this pattern isn't common in the rest of our codebase. I think it's important to maintain consistency in the codebase.